pkg/hnsw/hnsw.go

```diff
@@ -74,17 +75,25 @@ func (h *Hnsw) spawnLayer() int {
 searchLayer needs two things:
 1. todo! an item from a euc queue that computes the distance from the entry point node -> q.
 */
-func (h *Hnsw) searchLayer(q Vector, ef, layerId int, nearestNeighborsToQForEf *MaxQueue) {
+func (h *Hnsw) searchLayer(q Vector, entryNode *Node, ef int, layerId int, numNearestToQ *MinQueue) {
```

Comment:
@kevmo314 

`searchLayer` updates `numNearestToQ` BWO passing via params. Is this a good pattern?

Reply:
It's ok but not ideal. Generally to avoid this, the `Hnsw` struct should own that queue.

Reply:
> It's ok but not ideal. Generally to avoid this, the `Hnsw` struct should own that queue.

I'm a bit skeptical about making `Hnsw` own that queue, as these queues should be generated dynamically when we `Search` or `Insert`. I refactored the helper function to return a `MinQueue`, which I drain into the OG minqueue

Reply:
Yeah that approach works 👍